src/mbed_tools/cli/build.py

```diff
@@ -81,20 +81,18 @@ def build(
        sterm: Open a serial terminal to the connected target.
        baudrate: Change the serial baud rate (ignored unless --sterm is also given).
     """
+    _validate_target_and_toolchain_args(mbed_target, toolchain)
     if mbed_os_path is None:
         program = MbedProgram.from_existing(pathlib.Path(program_path))
     else:
         program = MbedProgram.from_existing(pathlib.Path(program_path), pathlib.Path(mbed_os_path))
-    mbed_config_file = program.files.cmake_config_file
     build_tree = program.files.cmake_build_dir
     if clean and build_tree.exists():
         shutil.rmtree(build_tree)
 
-    if any([not mbed_config_file.exists(), not build_tree.exists(), mbed_target, toolchain]):
-        click.echo("Configuring project and generating build system...")
-        _validate_target_and_toolchain_args(mbed_target, toolchain)
-        generate_config(mbed_target.upper(), toolchain, program)
-        generate_build_system(program.root, build_tree, profile)
+    click.echo("Configuring project and generating build system...")
```

Comment:
I'll revise this. Looking at it again it makes much more sense to drop the conditions where it will always generate with a new target and toolchain, rather than dropping it altogether, when supporting multiple targets.

Reply:
Looks we do indeed want to unconditionally configure, even when the build tree exists already. Apologies for the back and forth on this. We should make this change in a separate commit and explain the reasoning in the commit message.